webapp/portlet/src/main/webapp/vue-apps/space-settings/components/SpaceSettings.vue

```diff
@@ -26,18 +35,31 @@ export default {
     // add external components
     const externalComponents = extensionRegistry.loadComponents('external-space').map(component => component.componentOptions.componentImpl);
     this.spaceExternalSettings.push(...externalComponents);
-
-    document.addEventListener('addSpaceSettingsExternalComponents', (event) => {
-      if (event && event.detail) {
-        this.spaceExternalSettings.push(event.detail.componentImpl);
-      }
-    });
-
     document.addEventListener('hideSettingsApps', () => this.displayed = true);
     document.addEventListener('showSettingsApps', () => this.displayed = true);
   },
   mounted() {
+    document.addEventListener('addSpaceSettingsExternalComponents', (event) => {
+      if (event && event.detail && this.isApplicationInstalled(event.detail.appId)) {
+        this.spaceExternalSettings.push(event.detail.componentImpl);
+      }
+    });
     this.$nextTick().then(() => this.$root.$applicationLoaded());
   },
+  methods: {
+    isApplicationInstalled(appId) {
+      const allApplication = this.space && this.space.app.split(',');
```

Comment:
Let's use the idea you provided first and try to query the API and check if the application is installed with a Rest call

Reply:
Done !!